enclave/core/tracee.c

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include <openenclave/bits/safecrt.h>
+#include <openenclave/bits/safemath.h>
+#include <openenclave/bits/types.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/enclavelibc.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/sgxtypes.h>
+#include <openenclave/internal/trace.h>
+#include <openenclave/internal/utils.h>
+
+static oe_log_filter_t g_active_log_filter = {0};
+static char g_logging_filename[MAX_FILENAME_LEN];
+static bool g_debug_allowed_enclave = false;
+
+const char* get_filename_from_path(const char* path)
+{
+    if (path)
+    {
+        for (size_t i = oe_strlen(path) - 1; i >= 0; i--)
+        {
+            if (path[i] == '/')
+            {
+                return &path[i + 1];
+            }
+        }
+    }
+    return path;
+}
+
+// Read an enclave's identity attribute to see to if it was signed as an debug
+// enclave
+bool is_enclave_debug_allowed()
+{
+    bool ret = false;
+#ifdef OE_USE_LIBSGX
+
+#if defined(__linux__)
+    oe_result_t result = 0;
+    td_t* td = oe_get_td();
+    uint8_t* report_buffer = NULL;
+
+    if (td->simulate)
+    {
+        // enclave in simulate mode is treated as debug_allowed
+        ret = true;
+    }
+    else
+    {
+        uint64_t report_buffer_size = OE_MAX_REPORT_SIZE;
+        const sgx_report_t* sgx_report = NULL;
+        oe_report_header_t* header = NULL;
+
+        report_buffer = (uint8_t*)oe_malloc(OE_MAX_REPORT_SIZE);
+        if (report_buffer == NULL)
+            goto done;
+
+        // get a report on the enclave itself for enclave identity information
+        report_buffer_size = OE_MAX_REPORT_SIZE;
+        result = oe_get_report(
+            0, NULL, 0, NULL, 0, report_buffer, &report_buffer_size);
+        if (result != OE_OK)
+            goto done;
+
+        header = (oe_report_header_t*)report_buffer;
+        sgx_report = (const sgx_report_t*)header->report;
+        ret = (sgx_report->body.attributes.flags & SGX_FLAGS_DEBUG) != 0;
```

Comment:
Instead of using sgx_report, can you not use oe_report_t?

Reply:
Good point.
To do that, it would involve refactoring oe_parse_report from oehost to oecore, I tried that and it was pretty messy, thus I decided going with the current approach.